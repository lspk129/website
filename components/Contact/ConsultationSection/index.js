import styled from 'styled-components';

const ConsultationSection = styled.div`
  display: grid;
  grid-column: 2 / 3;
  justify-content: center;
`;

export default ConsultationSection;
