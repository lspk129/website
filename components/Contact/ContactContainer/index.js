import styled from 'styled-components';

const ContactContainer = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  height: 600px;
`;

export default ContactContainer;
