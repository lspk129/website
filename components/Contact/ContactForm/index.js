import styled from 'styled-components';

const ContactForm = styled.form``;

export default ContactForm;
