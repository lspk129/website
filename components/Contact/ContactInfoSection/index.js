import styled from 'styled-components';

const ContactInfoSection = styled.div`
  display: grid;
  grid-column: 1 / 2;
  justify-content: center;
`;

export default ContactInfoSection;
