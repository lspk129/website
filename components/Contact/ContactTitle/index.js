import styled from 'styled-components';

const ContactTitle = styled.h3`
  text-transform: uppercase;
`;

export default ContactTitle;
