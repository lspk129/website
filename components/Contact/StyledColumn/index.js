import styled from 'styled-components';
import { Column } from 'hedron';

const StyledColumn = styled(Column)`
  background: ${props => props.theme.secondaryBackground};
`;

export default StyledColumn;
