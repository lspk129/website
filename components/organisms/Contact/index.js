import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'react-styled-flexboxgrid';
import muiThemeable from 'material-ui/styles/muiThemeable';
import { TextField, SelectField, MenuItem, RaisedButton } from 'material-ui';
import { FaEnvelope, FaTwitter, FaGlobe } from 'react-icons/lib/fa';

import Heading from '../../atoms/Heading';
import IconWithLabel from '../../molecules/IconWithLabel';
import Division from '../../atoms/Division';
import ColStyled from '../../atoms/ColStyled';

class Contact extends Component {
  state = {
    value: '',
  };

  handleChange = (event, index, value) => this.setState({ value });

  render() {
    return (
      <Row center={'xs'}>
        <ColStyled xs={12} sm={6} color={this.props.muiTheme.palette.backgroundColor}>
          <Row center={'xs'}>
            <Col>
              <Division py={3}>
                <Division pb={1}>
                  <Heading upperCase>
                    Contact us
                  </Heading>
                </Division>
                <IconWithLabel
                  icon={<FaEnvelope color={this.props.muiTheme.palette.accent1Color} />}
                  label={'info@mokahaiku.com'}
                />
                <IconWithLabel
                  icon={<FaTwitter color={this.props.muiTheme.palette.accent1Color} />}
                  label={'@mokahaiku'}
                />
                <IconWithLabel
                  icon={<FaGlobe color={this.props.muiTheme.palette.accent1Color} />}
                  label={'www.mokahaiku.com'}
                />
              </Division>
            </Col>
          </Row>
        </ColStyled>

        <Col xs={10} sm={6}>
          <Division py={3}>
            <Division pb={1}>
              <Heading upperCase>
                Get a free consultation
              </Heading>
            </Division>
            <form>
              <Row center={'xs'}>
                <Col md={5} sm={5} xs={10}>
                  <Division px={1}>
                    <TextField floatingLabelText="Name" fullWidth />
                  </Division>
                </Col>
                <Col md={5} sm={5} xs={10}>
                  <Division px={1}>
                    <TextField floatingLabelText="Your email" fullWidth />
                  </Division>
                </Col>

                <Col xs={10}>
                  <Division px={1} textAlign={'left'}>
                    <SelectField
                      floatingLabelText="Choose a subject"
                      value={this.state.value}
                      onChange={this.handleChange}
                      fullWidth
                    >
                      <MenuItem value={1} primaryText="I want an introduction to your services..." />
                      <MenuItem value={2} primaryText="I want a free price quote..." />
                      <MenuItem value={3} primaryText="I want to ask a question" />
                    </SelectField>
                    {this.state.value === 3 && (
                    <TextField
                      floatingLabelText="Your question"
                      multiLine
                      rows={4}
                      fullWidth
                    />
                  )}
                  </Division>
                </Col>
              </Row>
              <Row center={'xs'}>
                <Col xs={10}>
                  <Division pl={1} pt={3} textAlign={'left'}>
                    <RaisedButton label={'Send'} secondary />
                  </Division>
                </Col>
              </Row>
            </form>
          </Division>
        </Col>
      </Row>
    );
  }
}

Contact.propTypes = {
  muiTheme: PropTypes.shape({
    palette: PropTypes.shape({
      accent1Color: PropTypes.string.isRequired,
      backgroundColor: PropTypes.string,
    }).isRequired,
  }).isRequired,
};

export default muiThemeable()(Contact);
