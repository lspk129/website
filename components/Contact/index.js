import React from 'react';

import ContactContainer from './ContactContainer';
import ContactInfoSection from './ContactInfoSection';
import ConsultationSection from './ConsultationSection';
import ContactTitle from './ContactTitle';
import ContactForm from './ContactForm';

const Contact = () => (
  <ContactContainer>
    <ContactInfoSection>
      <ContactTitle>
        Get a free consultation
      </ContactTitle>
    </ContactInfoSection>
    <ConsultationSection>
      <ContactForm />
    </ConsultationSection>
  </ContactContainer>
);

export default Contact;
