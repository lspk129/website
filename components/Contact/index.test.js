import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Contact from './';
import ContactContainer from './ContactContainer';
import ContactInfoSection from './ContactInfoSection';
import ConsultationSection from './ConsultationSection';
import ContactTitle from './ContactTitle';

describe('Contact', () => {
  const mountWrapper = mount(<Contact />);

  it('renders correctly', () => {
    const wrapper = renderer.create(<Contact />);
    const tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders ContactContainer component', () => {
    expect(mountWrapper.find(ContactContainer)).toBePresent();
  });

  it('renders ContactInfoSection component', () => {
    expect(mountWrapper.find(ContactInfoSection)).toBePresent();
  });

  it('renders ConsultationSection component', () => {
    expect(mountWrapper.find(ConsultationSection)).toBePresent();
  });

  it('renders text "Get free consultation"', () => {
    expect(mountWrapper.find(ContactTitle)).toHaveText('Get a free consultation');
  });
});
