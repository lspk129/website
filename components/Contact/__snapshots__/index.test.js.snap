// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders correctly 1`] = `
<div
  className="sc-bdVaJa gptHmk"
>
  <div
    className="sc-bwzfXH hchiPD"
  >
    <h3
      className="sc-bxivhb NHGCn"
    >
      Get a free consultation
    </h3>
  </div>
  <div
    className="sc-htpNat dGiqIK"
  >
    <form
      className="sc-ifAKCX cFlEyZ"
    />
  </div>
</div>
`;
